app/assets/stylesheets/base/boxes.scss

```diff
@@ -0,0 +1,40 @@
+@import '../vars/_variables.scss';
+
+@mixin box($this: $this, $color-1: $smoke-90, $color-2: $smoke-100) {
+  border-color: $color-2;
+
+  &#{$this}--filled {
+    background-color: $color-1;
+  }
+
+  @for $i from 1 through 3 {
+    &#{$this + '--level-' + $i} {
+      box-shadow: #{pow(2, $i) + px} #{pow(2, $i) + px} 0 $color-2;
+    }
+  }
+}
+
+.crayons-box {
+  $this: &;
```

Comment:
This line is so cool. Can you tell me what this means/does? 😂

Reply:
TL;DR - it's caching the current selector so in this particular case it helps me write shorter and better SCSS :) Actually - my explanation is really poor, go check out [this](https://css-tricks.com/snippets/sass/caching-current-selector-sass/).

Reply:
Woah, didn't know about this! maybe we can link to the article in the comments, or write a quick explanation!

Reply:
Good idea, i'll add the comment with link, just in case.